File: app/helpers/logic.rb
# Consolidates and simplifies commonly used logic

module LogicHelper

  def couple?

    if params[:user] == 'couple' || params[:user] == 'trial'
      true
    else
      false
    end

  end

  def guest?

    if params[:user] == 'guest'
      true
    else
      false
    end

  end

  def trial?

    if params[:user] == 'trial'
      true
    else
      false
    end

  end

# Extension of common logic to <body> classes for use in CSS
# ---------------------------------------------------------------------------

  def user
    if trial?
      'trial couple'
    elsif couple?
      'couple'
    else
      'guest'
    end
  end

end

File: public/prototype/logic.js
/* Logic Helper Bridge
----------------------------------------------------------------------------- */


    var guest;
    var trial;
    var couple;


    $(document).ready(function() {
    
        guest = ($('body').hasClass('guest') ? true : false);
        trial = ($('body').hasClass('trial') ? true : false);
        couple = ($('body').hasClass('couple') ? true : false);
    
    });